app/controllers/user_challenges_controller.rb

```diff
@@ -4,6 +4,7 @@ class UserChallengesController < ApplicationController
   before_action :ensure_user_exists!, only: [:create]
   before_action :ensure_challenge_not_full!, only: [:create]
   before_action :ensure_correct_user_for_delete!, only: [:destroy]
+  before_action :ensure_user_can_evaluate, only: [:evaluate_comments, :evaluate_places]
```

Comment:
Slight nit, other before_actions end in a `!`. I'm not sure what Rails convention dictates, but that's what I've been doing